### PR TITLE
[HyperShift MCE] enable MCE 2.5 and remove MCE 2.3 jobs for AWS

### PR DESCRIPTION
**Pull request by LiangquanLi930**

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-mce-conformance

**Review by rokej — APPROVED** [review]

File: ci-operator/config/openshift/hypershift/openshift-hypershift-release-4.13__periodics.yaml [hunk 1]
```diff
@@ -115,13 +115,6 @@ tests:
       TEST_TYPE: main
       VERIFY_COMPLIANCE_OPERATOR_SCAN: "true"
     workflow: hypershift-ibmcloud-e2e
-- as: e2e-aws-ovn-mce-conformance
```
Comment (jparrill) [inline ci-operator/config/openshift/hypershift/openshift-hypershift-release-4.13__periodics.yaml:118]: Looks like you're not only enabling MCE 2.5 in aws, also cleaning some stuff, could you please specify that in the PR/Commit?

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.13-periodics.yaml [hunk 2]
```diff
@@ -323,87 +323,6 @@ periodics:
     - name: result-aggregator
       secret:
         secretName: result-aggregator
-- agent: kubernetes
```
Comment (jparrill) [inline ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.13-periodics.yaml:326]: could you please specify a quick comment about why you're cleaning this job in the PR?

Comment (LiangquanLi930) [reply ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.13-periodics.yaml:326]: For this change, it is automatically generated by `Make job`

**Pull request merged**